docs/library/micropython.rst

```diff
@@ -147,3 +147,66 @@ Functions
 
    There is a finite queue to hold the scheduled functions and `schedule()`
    will raise a `RuntimeError` if the queue is full.
+
```

Comment:
@dpgeorge I've moved / updated the docs here... does this feel long enough now you'd prefer it in its own file?

Reply:
I think it's OK to keep it in this file.

If more classes get added in the future then we can factor them out.